### ISSUE TITLE
Unwanted dependencies of the Nuget package System.Net.Http.WinHttpHandler

### ISSUE THREAD

**Issue by elmodev001** [dotnet/runtime#31295]
I would like to use the Nuget package System.Net.Http.WinHttpHandler in order to have the HTTP/2 support on .NET Framwork 4.6.

I have installed it from nuget, and this make the packages manager to install a long list of dependencies:

```
<packages>
  <package id="Microsoft.NETCore.Platforms" version="1.1.0" targetFramework="net46" />
  <package id="Microsoft.Win32.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="NETStandard.Library" version="1.6.1" targetFramework="net46" />
  <package id="System.AppContext" version="4.3.0" targetFramework="net46" />
  <package id="System.Buffers" version="4.4.0" targetFramework="net46" />
  <package id="System.Collections" version="4.3.0" targetFramework="net46" />
  <package id="System.Collections.Concurrent" version="4.3.0" targetFramework="net46" />
  <package id="System.Console" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.Debug" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.DiagnosticSource" version="4.6.0" targetFramework="net46" />
  <package id="System.Diagnostics.Tools" version="4.3.0" targetFramework="net46" />
  <package id="System.Diagnostics.Tracing" version="4.3.0" targetFramework="net46" />
  <package id="System.Globalization" version="4.3.0" targetFramework="net46" />
  <package id="System.Globalization.Calendars" version="4.3.0" targetFramework="net46" />
  <package id="System.IO" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.Compression" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.Compression.ZipFile" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.FileSystem" version="4.3.0" targetFramework="net46" />
  <package id="System.IO.FileSystem.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Linq" version="4.3.0" targetFramework="net46" />
  <package id="System.Linq.Expressions" version="4.3.0" targetFramework="net46" />
  <package id="System.Memory" version="4.5.3" targetFramework="net46" />
  <package id="System.Net.Http" version="4.3.0" targetFramework="net46" />
  <package id="System.Net.Http.WinHttpHandler" version="4.6.0" targetFramework="net46" />
  <package id="System.Net.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Net.Sockets" version="4.3.0" targetFramework="net46" />
  <package id="System.ObjectModel" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Reflection.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Resources.ResourceManager" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.5.2" targetFramework="net46" />
  <package id="System.Runtime.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.Handles" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.InteropServices" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" targetFramework="net46" />
  <package id="System.Runtime.Numerics" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.Algorithms" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.Encoding" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.Primitives" version="4.3.0" targetFramework="net46" />
  <package id="System.Security.Cryptography.X509Certificates" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.Encoding" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.Encoding.Extensions" version="4.3.0" targetFramework="net46" />
  <package id="System.Text.RegularExpressions" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Tasks" version="4.3.0" targetFramework="net46" />
  <package id="System.Threading.Timer" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.ReaderWriter" version="4.3.0" targetFramework="net46" />
  <package id="System.Xml.XDocument" version="4.3.0" targetFramework="net46" />
</packages>
```

All is working fine on my side right now.

The problem is that I read that your suggestion is to NOT use the OOB System.Net.Http package, but I can't uninstall it from the nuget package manager, because it is a dependency of other packages.
So what is the correct way to use System.Net.Http.WinHttpHandler with the stock System.Net.Http?
Should I manually remove unwanted dependencies?
What is the minimal set of dependencies of System.Net.Http.WinHttpHandler?

**Comment by ViktorHofer** [comment]
@elmodev001 is it possible for you to upgrade to .NET Framework >= 4.6.1 or to .NET Core? We fixed this for 4.6.1 upwards.

**Comment by ViktorHofer** [comment]
Closing as we haven't heard back from the customer. In addition to that, we don't produce packages anymore that support .NET Framework < 4.6.1.
